### DISCUSSION TITLE
useData called on property change

### DISCUSSION THREAD

**Discussion by khchehab** [vikejs/vike#1544] [Discussion] [answered]
I added `+data.ts` to my page to fetch data to display in a table on the page from my endpoint. But my endpoint also supports pagination based on page and pageSize parameters.
Calling my endpoint with the initial (default) page and pageSize is good so far, but I am looking to make the data function also called when changing pages in my table, so I would have a state of the current page for example, and based on this state's update to fetch data with the `useData` hook.
I was thinking of using `PageContext`, but the documentation indicates that I should treat `PageContext` as immutable, is this even possible? Or will I have to implement the fetching myself without using the data hook?

My `+data.ts` file contains the data function and simply fetches the data from the api, which looks like this:
```typescript
export async function data() {
    const response = await fetch(myEndpointUrl, {
        method: 'GET'
    });
    const data = await response.json();
    // data.meta contains the current request's page and pageSize
    // data._links contains api urls for the previous and next page of the request
    return [ data.data, data.meta, data._links ];
}
```

And my `+Page.tsx` file contains a simple table in it with displaying the data from the request.

**Comment by brillout** [comment, marked as answer]
See 6b99bfd555d2820a6c7a7ce318f225aa8fd1d1a1 but let me know if you still have questions.

**Comment by khchehab** [comment]
Hi @brillout it seems I should be using API routes or RPC for my use case. I will check them out and try to apply them.
Thanks

**Comment by khchehab** [comment]
Just so that I know if I understood the use of telefunc in my case, I would keep the data hook to get the initial data for my page, but then for my navigation component (next and previous url) I would use the telefunc methods to fetch the data. Am I correct?

**Reply by brillout** [reply]
Correct.
